src/aggregate_score.rs

```diff
@@ -23,6 +25,18 @@ impl AggregateScore {
         self.four_of_a_kind += score.four_of_a_kind as u32;
         self.straight_flush += score.straight_flush as u32;
     }
+
+    pub fn add(&mut self, score: HandData<u32>) {
```

Comment:
`HandScore` is just `HandData<bool>`, so now we have two identical methods, but which differ only by the data type... time for generics!

I think the quick fix here is to make the existing `insert` method generic over the inner type `T` of `HandData`, rather than only working when `T` is a `bool`.

Reply:
But the 2 function are a little different. The add function function adds both high score but insert function adds 1 to the high score.